app/models/work.rb

```diff
@@ -10,20 +13,59 @@ class Work < ApplicationRecord
     end
   end
 
-  def self.create_skeleton(title, user_id, collection_id, work_type)
+  before_update do |ds|
+    if dublin_core.present?
+      # we don't mint ARKs for these records
+    elsif ds.ark.blank?
+      ds.ark = Ark.mint
+    end
+  end
+
+  after_save do |ds|
+    # We only want to update the ark url under certain conditions.
+    # Set this value in config/update_ark_url.yml
+    if Rails.configuration.update_ark_url
+      if ds.ark.present?
+        # Ensure that the ARK metadata is updated for the new URL
+        if ark_object.target != ds.url
+          ark_object.target = ds.url
+          ark_object.save!
+        end
+      end
+    end
+  end
+
+  validate do |ds|
+    if ds.ark.present?
+      ds.errors.add(:base, "Invalid ARK provided for the Work: #{ds.ark}") unless Ark.valid?(ds.ark)
+    end
+  end
+
+  def self.create_skeleton(title, user_id, collection_id, work_type, profile)
     work = Work.new(
       title: title,
       created_by_user_id: user_id,
       collection_id: collection_id,
       work_type: work_type,
-      state: "AWAITING-APPROVAL"
+      state: "AWAITING-APPROVAL",
+      profile: profile
     )
     work.save!
     work
   end
 
-  def dataset_id
-    Dataset.where(work_id: id).first&.id
+  # Convenience method to create Datasets with the DataCite profile
+  def self.create_dataset(title, user_id, collection_id)
+    work = Work.new(
+      title: title,
+      created_by_user_id: user_id,
+      collection_id: collection_id,
+      work_type: "DATASET",
+      state: "AWAITING-APPROVAL",
+      profile: "DATACITE"
+    )
+    work.save!
+    work
```

Comment:
This is just a convenience method (a sibling to the existing `created_skeleton()`) that defaults the parameters for Datacite datasets. Simplifies the calls to create a "dataset" which is really just a Work of type DATASET with the DATACITE profile.